src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -242,16 +274,25 @@ public sealed partial class Server : IDisposable {
             interpreter.ModuleResolution.ReloadAsync(cancellationToken).ContinueWith(t => {
                 _log?.Log(TraceEventType.Information, Resources.Done);
                 _log?.Log(TraceEventType.Information, Resources.AnalysisRestarted);
+
                 RestartAnalysis();
+                
+                if (_watchSearchPaths) {
+                    ResetPathWatcher();
+                }
             }, cancellationToken).DoNotWait();
 
         }
 
         private void RestartAnalysis() {
-            var analyzer = Services.GetService<IPythonAnalyzer>();;
+            var analyzer = Services.GetService<IPythonAnalyzer>();
             analyzer.ResetAnalyzer();
             foreach (var doc in _rdt.GetDocuments()) {
-                doc.Reset(null);
+                if (!doc.IsOpen) {
+                    _rdt.RemoveDocument(doc.Uri);
```

Comment:
I think you can `UnlockDocument` until count reaches 0 and then close it.

Reply:
If something has a lock on the document, shouldn't that mean that the other thing is what needs to do the unlocking? I do see `UnlockDocument` called at the beginning of ReloadAsync, though.

Reply:
Unlocking then closing wouldn't work with the current implementation of CloseDocument, actually. It requires that the document have at least one lock before preceding, and I had hit cases where closing documents put me into the situation where I'd hit the debug assert. Maybe I just allow it to close documents which have not been locked, but I feel like that assert was there for a reason.